src/libs/antares/study/parts/thermal/cluster.h

```diff
@@ -340,6 +340,32 @@ class ThermalCluster final : public Cluster, public std::enable_shared_from_this
     double marketBidCost = 0;
     //! Variable O&M cost (euros/MWh)
     double variableomcost = 0;
+
+    struct Ramping
+    {
+        //! Cost of power increase (euros/MW)
+        double powerIncreaseCost;
+        //! Cost of power decrease (euros/MW)
+        double powerDecreaseCost;
+        //! Maximum hourly upward power ramping rate (MW/hour)
+        double maxUpwardPowerRampingRate;
+        //! Maximum hourly downward power ramping rate (MW/hour)
+        double maxDownwardPowerRampingRate;
+
+        Ramping() :
+         powerIncreaseCost(0.),
+         powerDecreaseCost(0.),
+         maxUpwardPowerRampingRate(0.),
+         maxDownwardPowerRampingRate(0.)
+        {
+        }
+
+        void reset();
+        bool checkValidity(Area* area, Data::ClusterName clusterName);
+        friend std::ostream& operator<<(std::ostream&, const Ramping& ramping);
+    };
+    Ramping ramping;
```

Comment:
```suggestion
    std::optional<Ramping> ramping;
```
By default, the ramping constraints are not enabled for a given cluster. They are enabled only if an object exists within this `std::optional`.

Possible justification : there is no good default value for `maxUpwardPowerRampingRate` and `maxDownwardPowerRampingRate`, _ie_ we can't keep the existing behavior.

This induces a small difficulty in the variable/constraint counting process, in the sense that we must only take into account cluster for which ramping is enabled.

Reply:
Commit [8133150](https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/1817/commits/8133150b562b9e7655a86f76893116f5b9e5ccea) implements now is feature